kolibri/core/device/utils.py

```diff
@@ -141,3 +111,219 @@ def set_app_key_on_response(response):
+    if preset:
+        dataset_data = mappings[preset]
+        facility.dataset.preset = preset
+        for key, value in dataset_data.items():
+            setattr(facility.dataset, key, value)
+        facility.dataset.save()
+        logger.info("Facility preset changed to {preset}.".format(preset=preset))
+    return facility
+
+
+def setup_device_and_facility(
+    facility,
+    facility_name,
+    preset,
+    facility_settings,
+    device_settings,
+    username,
+    password,
+):
+    from kolibri.core.auth.models import FacilityUser
+
+    with transaction.atomic():
+        if facility is None and facility_name is not None:
+            facility = create_facility(
+                facility_name=facility_name,
+                preset=preset,
+            )
+
+            if facility_settings:
+                for key, value in facility_settings.items():
+                    setattr(facility.dataset, key, value)
+                facility.dataset.save()
+                logger.info(
+                    "Facility settings updated with {}".format(facility_settings)
+                )
+
+        provision_device(**device_settings)
+        logger.info("Device settings updated with {}".format(device_settings))
+
+        if username and password and facility:
+            try:
+                FacilityUser.objects.create_superuser(
+                    username, password, facility=facility
+                )
+                logger.info(
+                    "Superuser created with username '{username}' in facility '{facility}'.".format(
+                        username=username, facility=facility
+                    )
+                )
+            except ValidationError:
+                logger.warn(
+                    "An account with username '{username}' already exists in facility '{facility}', not creating user account.".format(
+                        username=username, facility=facility
+                    )
+                )
+
+
+def provision_from_file(file_path):
+    """
+    Expects a JSON file with the following format (example values supplied):
+    {
+        "facility": "My Facility",
+        "preset": "formal",
+        "facility_settings": {
+            "learner_can_edit_username": true,
+            "learner_can_edit_name": true,
+            "learner_can_edit_password": true,
+            "learner_can_sign_up": true,
+            "learner_can_delete_account": true,
+            "learner_can_login_with_no_password": true,
+            "show_download_button_in_learn": true
+        },
+        "device_settings": {
+            "language_id": "en",
+            "landing_page": "homepage",
+            "allow_guest_access": true,
+            "allow_peer_unlisted_channel_import": true,
+            "allow_learner_unassigned_resource_access": true,
+            "name": "My Device",
+            "allow_other_browsers_to_connect": true
+        },
+        "username": "superuser",
+        "password": "password"
```

Comment:
Slightly odd that `device_settings` and `facility_settings` have their own sub-objects, but these two don't.

E.g. 

```suggestion
        "super_user_credentials": {
            "username": "superuser",
            "password": "password"
        }
```

Do we have a principled way of deciding how deeply nested this file should be, and when nesting is appropriate? If not, we might want to consider imposing some rules.

Reply:
I think this seems like a good change, yes.